ARExperiment/ARExperiment/LightsAnimationsViewController.swift

```diff
@@ -47,36 +36,24 @@ class LightsAnimationsViewController: UIViewController, ARExperimentSessionHandl
         sceneView.session.pause()
     }
     
-    private func setUpModelsOnLoad() {
-        earthNode = modelForNodeName(earthNodeName)
-        moonNode = modelForNodeName(moonNodeName)
-        planeNodeModel = modelForNodeName(planeNode)
-        lightNodeModel = modelForNodeName(lightNode)
-    }
-    
-    private func modelForNodeName(_ nodeName: String) -> SCNNode? {
-        return arModel.createSceneNodeForAsset(nodeName,
-                                               assetFolder: assetFolder,
-                                               fileName: fileName,
-                                               assetExtension: fileExtension)
-    }
-    
     override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
         guard let location = touches.first?.location(in: sceneView) else {
             return
         }
-        
-        if let nodeExists = sceneView.scene.rootNode.childNode(withName: earthNodeName, recursively: true),
-            let secondNodeExists = sceneView.scene.rootNode.childNode(withName: moonNodeName, recursively: true) {
-            nodeExists.removeFromParentNode()
-            secondNodeExists.removeFromParentNode()
+
+        for node in arAsset.nodes {
+            if let nodeExists = arModel.node(in: sceneView, named: node.name) {
```

Comment:
nit - I'd call the variable `existingNode` - `nodeExists` sounds like it might be a boolean

Reply:
is actually the node if that one exists, so we then remove it. But I like the name change, makes more sense.